File: 234/palindrome-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public boolean isPalindrome(ListNode head) {
        ListNode head2 = null;
        ListNode now = head;
        ListNode now2 = null;
        while(now != null){
            head2 = new ListNode(now.val, head2);
            now = now.next;
        }
        now = head;
        now2 = head2;
        while(now != null){
            if(now.val!=now2.val) return false;
            now = now.next;
            now2 = now2.next;
        }
        return true;
    }
}

File: 234/palindrome-linked-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
bool isPalindrome(struct ListNode* head){
    int a[100001], N=0;
    while(head!=NULL){
        a[N++] = head->val;
        head = head->next;
    }
    for(int i=0; i<N/2; i++){
        if(a[i]!=a[N-1-i]) return false;
    }
    return true;
}
